hack/unit.sh

```diff
@@ -11,9 +11,15 @@ if [ "${IS_CONTAINER}" != "false" ]; then
   eval "$(go env)"
   cd "${GOPATH}"/src/github.com/metal3-io/baremetal-operator
   export XDG_CACHE_HOME="/tmp/.cache"
-
   export COVER_PROFILE="${ARTIFACTS}"/cover.out
-
+  
+  # The following downloads a set of envtest binaries to run tests with
+  export K8S_VERSION=1.22.1
+  curl -sSLo envtest-bins.tar.gz "https://go.kubebuilder.io/test-tools/${K8S_VERSION}/$(go env GOOS)/$(go env GOARCH)"
+  mkdir /usr/local/kubebuilder
+  tar -C /usr/local/kubebuilder --strip-components=1 -zvxf envtest-bins.tar.gz
```

Comment:
What is this stuff actually for? Can we build from source in the `hack` module the way we do for other stuff?

Is this file even used by anything?

Reply:
unit.sh is used by [unit test job in Prow](https://github.com/metal3-io/project-infra/blob/b124cfa18a06e27e78cc7c1b86da4ce4d58433f0/prow/config/config.yaml#L281) on a pull request.

We need to download these tarball, because it contains all necessary binaries like etcd, kubelet, api-server and etc, which we need during the test - to set up and start etcd instance and the k8s API server.

Reply:
The new suite test file which checks for example if apis are added correctly to schemas, require the binaries and I have followed the kubebuilder way of doing envtest from `https://book.kubebuilder.io/reference/envtest.html`. As long as the file's usage as @fmuyassarov indicated we do use them in prow. I am open to ideas if there are other alternatives to do this.

Reply:
FWIW kubebuilder has been generating the suite_test file for a while now but we keep deleting it because it doesn't work (at least locally, not sure about CI). Perhaps that's because it was missing this stuff... but also this script isn't called when we run tests locally so that makes me suspect this won't work either.
TBH I'd be happy to just keep deleting the suite_test every time kubebuilder adds it. We can be reasonably confident that autogenerated code works :)

Reply:
Alright, I am not so strong about this either. I will push an update soon removing suite_test